### PR TITLE
osd: use cp -a command for copying init-containers binary

### LINKED ISSUE #5296: init-copy-binaries can leave container in a bad state
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
If the program crashes (due to OOM, segfault, etc), the restarting the container should not leave things in a bad state

**Expected behavior:**
If the copy fails, the copy is re-attempted on next init container run

**How to reproduce it (minimal and precise):**

1. Set a limit range on the namespace:
```
apiVersion: v1
kind: LimitRange
metadata:
  name: rook-ceph
  namespace: rook-ceph
spec:
  limits:
  - type: Container
    defaultRequest:
      memory: 8Mi
    default:
      memory: 64Mi
```
2. Restart rook-ceph-operator so it triggers the `rook-ceph-csi-detect-version` job
3. Watch the job get OOMKilled then restarted

**Environment**:

Rook Version 1.3.1
Ceph CSI Version 2.0.1

**Details**:
Looking at the code, it looks like you don't do an atomic copy and skip if the file already exists. So when the program crashes, the incomplete file with the wrong permissions is left there so it manifests as a "Permission Denied" when trying to run the executable without the executable bits set.

Aside from the issue above, why does copying files take over 64MiB of memory!? Can't you just make this a `cp -a` instead of implementing a simple function like this in Go?

### PR DESCRIPTION
**Pull request by Sheetalpamecha**
remove copy-binaries cmd and related code

**Description of your changes:**

**Which issue is resolved by this Pull Request:**
Resolves #5296 

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure).
- [ ] **Skip Tests for Docs**: If this is only a documentation change, add the label `skip-ci` on the PR.
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

File: pkg/operator/k8sutil/cmdreporter/cmdreporter.go [hunk 1]
```diff
@@ -333,11 +333,14 @@ func (cr *cmdReporterCfg) initContainers() []v1.Container {
...
-		Name: CopyBinariesInitContainerName,
-		// Command: the rook command is the default entrypoint of rook images already
+		Name:    CopyBinariesInitContainerName,
+		Command: []string{"cp"},
 		Args: []string{
-			"copy-binaries",
-			"--copy-to-dir", CopyBinariesMountDir,
+			"--archive",
+			"--force",
+			"--verbose",
```
Comment (galexrt) [inline pkg/operator/k8sutil/cmdreporter/cmdreporter.go:341]: How many lines does the verbose flag cause on the `cp` command in comparison to the current copy binaries command?

**Review by travisn — COMMENTED** [review]
@BlaineEXE Do you recall why we implemented the `copy-binaries` command instead of using `cp` like this?

**Comment by BlaineEXE** [comment]
I don't recall why I implemented it as a daemon. It used to copy `tini` as well as `rook`, but that's no longer needed. I think I may have implemented it as a binary simply because the behavior used to be accomplished by the OSD daemon process and not considered `cp` as a good method. TBH, I think this is a much better approach.

**Comment by BlaineEXE** [comment]
This is not urgent, so I will wait for @travisn to get back from vacation. I think it may make sense to merge this into master and not backport it so that we have some time to make sure it does not negatively affect anything in master. I don't anticipate any issues, but there is always a small chance.

File: pkg/operator/k8sutil/cmdreporter/cmdreporter.go (hunk 1 above)
Comment (travisn) [reply pkg/operator/k8sutil/cmdreporter/cmdreporter.go:341]: @Sheetalpamecha How verbose is this flag? Do you have test output from the container?

Comment (Sheetalpamecha) [reply pkg/operator/k8sutil/cmdreporter/cmdreporter.go:341]: > How many lines does the verbose flag cause on the `cp` command in comparison to the current copy binaries command?

@galexrt Both ways container prints one line only. 
@travisn Here are both the outputs - 

with copy binaries cmd - 

```
sheetalpamecha@Sheetals ~/rook (init-copy-binaries)> kubectl -n rook-ceph logs rook-ceph-osd-prepare-minikube-n884d -c copy-bins
2023-07-31 04:02:15.507266 I | job-reporter-cmd: copying /usr/local/bin/rook to /rook/rook
```

with cp cmd -
```
runner@fv-az625-580:~/work/rook/rook$ k logs rook-ceph-osd-prepare-fv-az625-580-b6564 -c copy-bins
'/usr/local/bin/rook' -> '/rook/rook'
```
If we remove verbose flag nothing is printed in logs.

**Comment by Sheetalpamecha** [comment]
No code change done, forced pushed to rerun all failing test case.

**Pull request merged**